src/main/java/org/jfree/chart/axis/CyclicNumberAxis.java

```diff
@@ -476,7 +479,9 @@ protected List refreshTicksHorizontal(Graphics2D g2, Rectangle2D dataArea,
                         && (currentTickValue != cycleBound)) {
                     anchor = isInverted()
                         ? TextAnchor.TOP_RIGHT : TextAnchor.BOTTOM_RIGHT;
-                    result.remove(result.size() - 1);
+                    @SuppressWarnings("index") // If cycled, there must be at least one tick in the results list already

```

Comment:
This appears a number of times.  You could abstract it out as a new private method `removeLast`.
I think that would be clearer, it would reduce the size of the diffs, and it would reduce the number of warning suppressions.